core/app/[locale]/(default)/webpages/contact/[id]/page.tsx

```diff
@@ -1,54 +1,189 @@
+  const webpage = await getWebPage(id);
+
+  return {
+    ...webpage,
+    content: message,
+  };
+}
+
+async function getContactFields(id: string) {
+  const t = await getTranslations('WebPages.ContactUs.Form');
+  const { entityId, path, contactFields } = await getWebPage(id);
+  const toGroupsOfTwo = (fields: Field[]) =>
+    fields.reduce<Array<FieldGroup<Field>>>((acc, _, i) => {
+      if (i % 2 === 0) {
+        acc.push(fields.slice(i, i + 2));
+      }
+
+      return acc;
+    }, []);
+
+  const pageIdField: Field = {
+    id: 'pageId',
+    name: 'pageId',
+    type: 'hidden',
+    defaultValue: String(entityId),
+  };
+
+  // Used for redirect to self with query params
+  const pagePathField: Field = {
+    id: 'pagePath',
+    name: 'pagePath',
+    type: 'hidden',
+    defaultValue: path,
+  };
+
+  const emailField: Field = {
+    id: 'email',
+    name: 'email',
+    label: `${t('emailLabel')} *`,
+    type: 'email',
+    required: true,
+  };
+
+  const commentsField: Field = {
+    id: 'comments',
+    name: 'comments',
+    label: `${t('commentsLabel')} *`,
+    type: 'textarea',
+    required: true,
+  };
+
+  const optionalFields = contactFields
+    .filter((field): field is ContactField => Object.hasOwn(fieldMapping, field))
+    .map<Field>((field) => ({
+      id: field,
+      name: field,
+      label: t(fieldMapping[field]),
+      type: 'text',
+      required: false,
+    }));
+
+  return [
+    ...toGroupsOfTwo([emailField, ...optionalFields]),
+    commentsField,
+    pageIdField,
+    pagePathField,
+  ];
+}
+
+export async function generateMetadata({ params }: Props): Promise<Metadata> {
+  const { id } = await params;
+  const webpage = await getWebPage(id);
   const { pageTitle, metaDescription, metaKeywords } = webpage.seo;
 
   return {
-    title: pageTitle,
+    title: pageTitle || webpage.title,
     description: metaDescription,
-    keywords: metaKeywords,
+    keywords: metaKeywords ? metaKeywords.split(',') : null,
   };
 }
 
-export default async function WebPage({ params }: Props) {
+export default async function ContactPage({ params, searchParams }: Props) {
   const { id } = await params;
+  const { success } = await searchParams;
+  const t = await getTranslations('WebPages.ContactUs.Form');
 
-  const data = await getWebpageData({ id: decodeURIComponent(id) });
-  const webpage = data.node?.__typename === 'ContactPage' ? data.node : null;
+  // TODO: Use reCaptcha
+  // const recaptchaSettings = await bypassReCaptcha(data.site.settings?.reCaptcha);
 
-  if (!webpage) {
-    notFound();
+  if (success === 'true') {
+    return (
+      <WebPageContent
+        breadcrumbs={getWebPageBreadcrumbs(id)}
+        webPage={getWebPageWithSuccessContent(id, t('success'))}
+      >
+        <div className="mx-auto w-full max-w-4xl">
+          <ButtonLink className="mt-8" href="/" size="large" type="submit" variant="primary">
+            {t('successCta')}
+          </ButtonLink>
+        </div>
```

Comment:
We should avoid defining CSS like this in Catalyst, otherwise we won't we able to swap out VIBES while keeping things visually consistent